systests/jaxrs/src/test/java/org/apache/cxf/systest/jaxrs/JAXRSClientServerBookTest.java

```diff
@@ -926,15 +926,12 @@ public void testCachingExpiresUsingETag() throws Exception {
         // Now make a second call. The value in the clients cache will have expired, so it should call
         // out to the service, which will return 304, and the client will re-use the cached payload
```

Comment:
Seems like ` ... and the client will re-use the cached payload` is not correct anymore?

Reply:
I took that to mean that the client should keep on using the same `Book` entity that it read from the first Response, but now that you mention it, it isn't very clear.  

I think we can either update the text of the comment to indicate that the client process should re-use the previously obtained entity, or we could add `response.bufferEntity()` and then pull the entity from the original response - something like this:

```
        Response response2 = target.request().get();
        assertEquals(304, response2.getStatus());
        assertFalse(response2.hasEntity());
        Book book2 = response.readEntity(Book.class);
        assertEquals(123L, book2.getId());
        cacheControlFeature.close();
```

Wdyt?

Reply:
👍 to adding these assertions, thanks!